### PR TITLE
Fixed typing on IntervalIndex functions

### PR DESCRIPTION
**Pull request by amotzop**
* Added typing to some of the methods of `IntervalIndex`.
* Removed `IntervalIndex` methods that are inherited from `Index`.
* Tests added: `test_interval_index.py` checking the function `from_breaks`, `from_arrays`, `from_tuples` and `is_overlapping`.

### CODE REVIEW

File: pandas-stubs/core/indexes/interval.pyi [hunk 1]
```diff
@@ -19,38 +24,46 @@ class IntervalIndex(IntervalMixin, ExtensionIndex):
     def __new__(
         cls,
         data,
-        closed=...,
-        dtype=...,
+        closed: Literal["left", "right", "both", "neither"] = ...,
```
Comment (twoertwein) [inline pandas-stubs/core/indexes/interval.pyi:27]: Could use `IntervalClosedType` from `pandas._typing`

Comment (Dr-Irv) [reply pandas-stubs/core/indexes/interval.pyi:27]: And elsewhere in this PR as well to use `IntervalClosedType`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Aside from comment to use `IntervalClosedType`, need to fix style issues that cause CI to fail

**Review by twoertwein — APPROVED** [review]

**Review by Dr-Irv — APPROVED** [review]
Thanks @amotzop

**Pull request merged**